crates/common/mqtt_channel/src/connection.rs

```diff
@@ -234,7 +243,12 @@ impl Connection {
                 // but will immediately be returned by `event_loop.poll()`
                 biased;
 
-                event = event_loop.poll() => event,
+                event = event_loop.poll() => {
+                    // If we receive more events, we want to reset the disconnect permit
+                    // so the event loop doesn't block disconnection
+                    disconnect_permit.take();
```

Comment:
Does that really help? After the permit is acquired in line no 254 during an iteration, we start the next iteration and would have triggered the disconnection in line no. 235. Since this `event_loop.poll()` happens after that, setting it back to None doesn't really help, does it? As the damage (triggering the disconnect task) has already been done.

Reply:
I thought there was a chance it might not since we definitely have cases where we acquire the disconnect permit, but then can't send the disconnect due to pending packets, but if we have pending packets that means we have to wait for event_loop.poll() to complete before we can achieve the required precondition for sending the disconnect. So thinking about it more thoroughly, I think you're correct, this is redundant.